src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -386,7 +387,34 @@ public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilde
             );
         }
     }
-
+    public static class RenderedPolyface extends RenderedShape<ShapeDispatcher.Polyface>
+    {
+        public RenderedPolyface(Minecraft client, ShapeDispatcher.ExpiringShape shape)
+        {
+            super(client, (ShapeDispatcher.Polyface)shape);
+        }
+        @Override
+        public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, double cx, double cy, double cz, float partialTick)
+        {
+                bufferBuilder.begin(switch(shape.mode){
+                    case 0->VertexFormat.Mode.TRIANGLE_FAN;
+                    case 1->VertexFormat.Mode.TRIANGLE_STRIP;
+                    case 2->VertexFormat.Mode.TRIANGLES;
+                    default -> throw new IllegalArgumentException("Unexpected value: " + shape.mode);}, DefaultVertexFormat.POSITION_COLOR);
+                for(int i=0;i<shape.vertex_list.size();i++){
+                    Vec3 vec=shape.vertex_list.get(i);
+                    if(shape.relative.get(i)){
+                        vec=shape.relativiseRender(client.level, vec, partialTick);
+                    }
+                    bufferBuilder.vertex(vec.x()-cx, vec.y()-cy, vec.z()-cz).color(shape.fr, shape.fg, shape.fb, shape.fa).endVertex();
+                }
+                tessellator.end();
+                
+        }
+        @Override
+        public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilder builder, double cx, double cy,
```

Comment:
Don't forget to render lines too - they accept different colors and should be rendered independently from faces.

Reply:
done.